File: autocompletetextview/MainActivity.java
package com.jixiaoyong.autocompletetextview;

import android.app.Activity;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.AutoCompleteTextView;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;

/*
* 2016年4月23日 11:24:51
* AutoCompleteTextView自动填充文本的使用
* */
public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //获得AutoCompleteTextView对象
        AutoCompleteTextView autoCompleteTextView = (AutoCompleteTextView)findViewById(R.id.auto_complete_text_view);

        //预设的联想字符数组，注意这里是ArrayList而不是List
        List<String> list = new ArrayList<String>();
        list.add("just for test1");
        list.add("just for test2");
        list.add("just for test3");
        list.add("just for test4");
        list.add("just for test5");

        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1,list);
        //将ArrayAdapter设置给autoCompleteTextView
        autoCompleteTextView.setAdapter(adapter);
    }
}

File: project/MyKotlin/app/src/main/java/cf/android666/mykotlin/MainActivity.kt
package cf.android666.mykotlin

import android.app.FragmentTransaction
import android.content.Context
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.design.widget.BottomNavigationView
import android.support.v4.view.PagerAdapter
import android.support.v4.view.ViewPager
import android.view.MenuItem
import android.widget.Toast
import cf.android666.mykotlin.fragment.Fragment2
import cf.android666.mykotlin.fragment.Fragment3
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.activity_main.view.*
import java.util.ArrayList

class MainActivity : AppCompatActivity() {

    var mContext: Context = this

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        var arrayList : ArrayList<BaseFragment>  = arrayListOf()

        arrayList.add(Fragment1())
        arrayList.add(Fragment2())
        arrayList.add(Fragment3())

        var fm  = supportFragmentManager

        view_pager.adapter = MPagerAdapter(arrayList,fm)

        bottom_nav.setOnNavigationItemReselectedListener(
                {
                    var id = it.itemId
                    when (id) {
                        R.id.menu1 -> view_pager.currentItem = 1
                        R.id.menu2 -> {toast("menu2");view_pager.currentItem = 2

                        }
                        R.id.menu3 -> {toast("menu3");view_pager.currentItem = 3
                        }
                    }

                }
        )

    }

    private fun toast(s: String) {
        Toast.makeText(mContext, s, Toast.LENGTH_SHORT).show()
    }
}
